webapp/graphite/render/evaluator.py

```diff
@@ -104,9 +104,25 @@ def evaluateTokens(requestContext, tokens, replacements=None, pipedArg=None):
 
     def handleInvalidParameters(e):
       if not getattr(handleInvalidParameters, 'alreadyLogged', False):
+        logMsg = str(e)
+
+        if 'sourceIdHeaders' in requestContext:
+          if 'HTTP_X_PANEL_ID' in requestContext['sourceIdHeaders']:
+            logMsg = 'panel-id: {id}; {msg}'.format(
+              id=requestContext['sourceIdHeaders']['HTTP_X_PANEL_ID'],
+              msg=logMsg)
+          if 'HTTP_X_DASHBOARD_ID' in requestContext['sourceIdHeaders']:
+            logMsg = 'dashboard-id: {id}; {msg}'.format(
+              id=requestContext['sourceIdHeaders']['HTTP_X_DASHBOARD_ID'],
+              msg=logMsg)
+          if 'HTTP_X_GRAFANA_ORG_ID' in requestContext['sourceIdHeaders']:
+            logMsg = 'org-id: {id}; {msg}'.format(
+              id=requestContext['sourceIdHeaders']['HTTP_X_GRAFANA_ORG_ID'],
+              msg=logMsg)
+
```

Comment:
Any particular reason for prepending rather than appending the new items to the message?

Reply:
not really, i just thought it looks better in the message

Reply:
what format do you think would look better?

Reply:
my initial thought was that adding it at the end of the log message would keep the messages more consistent if you're scanning through logs

Reply:
how about:

```
2020-03-09,20:49:44.345 :: Received invalid parameters (Invalid "float" value specified for function "scale" parameter "factor": u'0.123'): scale ([], u'0.123'); source: (org-id: 1; dashboard-id: 1; panel-id: 2)
```

And if the source headers aren't set the last part simply doesn't get appended:

```
2020-03-09,20:49:44.345 :: Received invalid parameters (Invalid "float" value specified for function "scale" parameter "factor": u'0.123'): scale ([], u'0.123')
```

Reply:
sounds reasonable to me

Reply:
updated it and added unit tests for that func which generates the message